lib/javascript/JavascriptModulesPlugin.js

```diff
@@ -692,7 +695,7 @@ class JavascriptModulesPlugin {
 			chunkGraph.getOrderedChunkModulesIterableBySourceType(
 				chunk,
 				"javascript",
-				compareModulesByIdentifier
+				compareModulesByReadableIdentifier(compilation.requestShortener)
```

Comment:
This will break the caching, since you pass a different comparator every time.

Reply:
You can do that my caching the comparator function with a WeakMap. There are some examples in the comparators file.